cajita/unit_test/tstSparseIndexSpace.hpp

```diff
@@ -0,0 +1,681 @@
+                    cell_num_per_tile_dim, cell_num_per_tile, HashType,
+                    key_type, value_type>
+        bis( size, size, size, capacity, rehash_factor );
+
+    Kokkos::View<int[size][size][size], TEST_DEVICE> i_res( "i" );
+    Kokkos::View<int[size][size][size], TEST_DEVICE> j_res( "j" );
+    Kokkos::View<int[size][size][size], TEST_DEVICE> k_res( "k" );
+    TEST_EXECSPACE().fence();
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<TEST_EXECSPACE>( 0, size ), KOKKOS_LAMBDA( int i ) {
+            for ( int j = 0; j < size; j++ )
+                for ( int k = 0; k < size; k++ )
+                {
+                    bis.insert( i, j, k );
+                    auto tile_key = bis.ijk2key( i, j, k );
+                    int ti, tj, tk;
+                    bis.key2ijk( tile_key, ti, tj, tk );
+                    i_res( i, j, k ) = ti;
+                    j_res( i, j, k ) = tj;
+                    k_res( i, j, k ) = tk;
+                }
+        } );
+    TEST_EXECSPACE().fence();
+    auto i_mirror =
+        Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), i_res );
+    auto j_mirror =
+        Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), j_res );
+    auto k_mirror =
+        Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), k_res );
+    for ( int i = 0; i < size; i++ )
+        for ( int j = 0; j < size; j++ )
+            for ( int k = 0; k < size; k++ )
+            {
+                EXPECT_EQ( i_mirror( i, j, k ), i );
+                EXPECT_EQ( j_mirror( i, j, k ), j );
+                EXPECT_EQ( k_mirror( i, j, k ), k );
+            }
+    bool test[size * size * size];
+
+    Kokkos::View<int[size * size * size], TEST_DEVICE> id_find_res( "id_find" );
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<TEST_EXECSPACE>( 0, size ), KOKKOS_LAMBDA( int i ) {
+            for ( int j = 0; j < size; j++ )
+                for ( int k = 0; k < size; k++ )
+                {
+                    auto find_id = bis.find( i, j, k );
+                    id_find_res( i * size * size + j * size + k ) = find_id;
+                }
+        } );
+    auto id_find_mirror =
+        Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), id_find_res );
+    for ( int i = 0; i < size; i++ )
+        for ( int j = 0; j < size; j++ )
+            for ( int k = 0; k < size; k++ )
+            {
+                test[i * size * size + j * size + k] = false;
+            }
+    for ( int i = 0; i < size; i++ )
+        for ( int j = 0; j < size; j++ )
+            for ( int k = 0; k < size; k++ )
+            {
+                test[id_find_mirror( i * size * size + j * size + k )] = true;
+            }
+    for ( int i = 0; i < size; i++ )
+        for ( int j = 0; j < size; j++ )
+            for ( int k = 0; k < size; k++ )
+            {
+
+                EXPECT_EQ( test[i * size * size + j * size + k], true );
+            }
+}
+
+void testSparseIndexSpaceFullInsert()
```

Comment:
Need to test `capacity()`, `size()`, and `reserve()` in this test!

Reply:
Added